### PR TITLE
Sales rep accounts view add category selectors

### PR DESCRIPTION
**Pull request by soft-Dev**
Added view selector with dynamic column selection from Customer table. Column value is set in ngOnInit.

### CODE REVIEW

**Comment by bitingmidge** [comment]
@soft-Dev Something seems wrong here but I'm not sure if I'm missing something. First thing I noticed was a lot of changes in the GetAccounts.sql file that looked like SQL being changed that doesn't have to do with category selectors.

Can you take a look at that file's diff and see if it looks correct --> https://github.com/bezlio/bezlio-apps/pull/268/files#diff-5285dfcf726ac2d78c7be384cc6ab1d8

**Comment by soft-Dev** [comment]
@bitingmidge It seems I put a blank line at the top accidentally which threw off parts of it. The only added bit here is the custom column, {Col} as FilterValue, which allows users to specify a custom filter column. Looks good to commit to me.

**Comment by bitingmidge** [comment]
@soft-Dev I think we must not be looking at the same thing. I'll send you some screenshots in my response to Adam's email.

**Comment by soft-Dev** [comment]
Ahh, sorry, I believe I see what happened now. I pulled the old GetAccounts.sql from my BRDB install and overwrote the latest that was on git. Let me fix that and push again.

**Comment by bitingmidge** [comment]
Okay, that makes sense. I think we have something similar with the BezlExport.txt file. I see some recent changes that are going to be lost with the pull request. You may need to import a fresh copy of the bezl from development (import from portal-dev) and implement the changes there again.

**Comment by soft-Dev** [comment]
Good point, didn't think of that either. I'll pull that down and add that back in. I'm working on the Map View category now, so this will probably happen later.

File: roles/sales-rep/accounts-view/js/onDataChange.js [hunk 1]
```diff
@@ -52,6 +52,11 @@ define(["./account.js"], function (account) {
             }
         }
 
+        //account filter types
+        if (bezl.data.Accounts && bezl.data.Categories) {
+            console.log(bezl.data.Categories);
```
Comment (bitingmidge) [inline roles/sales-rep/accounts-view/js/onDataChange.js]: Is this console.log() call supposed to be here?

Comment (soft-Dev) [inline roles/sales-rep/accounts-view/js/onDataChange.js]: Nope, just removed it.

**Comment by soft-Dev** [comment]
@bitingmidge Pulled the latest BezlExport and added my changes back in.

File: roles/sales-rep/accounts-view/BezlExport.txt [hunk 2]
```diff
@@ -600,15 +605,27 @@
...
+      "code": "bezl.vars.config = {};\n\nbezl.vars.config.baseUrl = \"\";\nif(bezl.vars.config.baseUrl == \"\") {\n  \tif(window.location.href.indexOf(\"portal-dev.bezl.io\") !== -1) {\n     \tbezl.vars.config.baseUrl = \"https://bezlio-apps-dev.bezl.io\";\n    } else {\n     \tbezl.vars.config.baseUrl = \"https://bezlio-apps.bezl.io\"; \n    }\n}\n\nbezl.vars.config.baseUrl = \"https://rawgit.com/bezlio/bezlio-apps/Sales-Rep-Accounts-View-Add-Category-Selectors\";\n\nbezl.vars.config.baseJsUrl = bezl.vars.config.baseUrl + \"/roles/sales-rep/accounts-view/js/\";\nbezl.vars.config.baseLibraryUrl = bezl.vars.config.baseUrl + \"/libraries/\";",
```
Comment (bitingmidge) [inline roles/sales-rep/accounts-view/BezlExport.txt]: @soft-Dev I'm sorry to create more work for you but can you take out the RawGit line in the setConfig() function?

**Comment by soft-Dev** [comment]
@bitingmidge No problem. Removed the RawGit line from the Bezl.

**Comment by bitingmidge** [comment]
@soft-Dev Thanks! It looks good to me, however, I'm going to leave this open for the moment until we can get some clarification from @aellis about the following he had in a separate email:

1. A given account can have one or many categories or tags associated with them.
2. Add a category selector to the top of the screen for accounts view to allow the quick filter using these categories.  I believe the default should be all are active initially.

**Comment by bitingmidge** [comment]
Closes #193

**Pull request merged**